src/libs/notifications.js

```diff
@@ -27,7 +27,7 @@ export const notify = (type, title, props) => {
   const notification = makeNotification({ type, title, ...props });
   if (!isNotificationMuted(notification.id)) {
     const visibleNotificationIds = _.map('id', notificationStore.get());
-    notificationStore.update(Utils.append(notification));
+    notificationStore.update((currentNotificationStore) => append(notification, currentNotificationStore));
```

Comment:
The previous implementation required the reader's knowledge that [notificationStore is of type atom](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/state.ts#L252) and in turn that [atom's update function expects a mutator function that will be passed the current value of the atom](https://github.com/DataBiosphere/terra-ui/blob/dev/packages/core-utils/src/state-utils.ts#L47).

I believe that my change above is more readable by making the input explicit (currentNotificationStore) and pass both values to the append function.

Reply:
In my opinion, without the curry, there's little reason for `append` to exist. I'll claim this would be even more readable inlined.

```suggestion
    notificationStore.update((previousNotifications) => [...previousNotifications, notification]));
```

Also, nit on naming. The value passed to the update function is not the store itself, but the value contained in the store.

Reply:
That's a fair point, but in either case, I don't think that currying was adding value here.

I would agree that inline would be more readable, but when I first started this work, it sounded as though there were people who preferred using the function. 

I like your naming suggestion. Thanks for that.

Reply:
@msilva-broad @cahrens do you have thoughts on removing append in favor of inlining?

Reply:
I think there's definitely a case for the _curried_ function vs inlining since it allows writing updates much more concisely.

Reply:
I may have missed an important meeting, but as a fan of curried functions I have to disagree that the newer form is more "readable". Readability is subjective of course, but for me, the original form was easier to read and write.

I'd prefer to continue writing code in a curried style where possible. I suppose it's fine to have some parts use a different form, but I'd advocate against removing curried forms everywhere, or removing the APIs that support curried usages like this.

Reply:
Good point @pshapiro4broad. I agree that the readability is probably a matter of subjectivity.

Reply:
I have to come to appreciate curried functions for updates like this or React's `setState`, where you pass a function that transforms the current value to a new value.

On the other hand, there is a learning curve for that FP style. Generally, I'd say that we have been trending away from an FP style in the interest of making the Terra UI codebase more easily approachable. Along those same lines, there's also the shift to JSX instead of React Hyperscript Helpers.

Reply:
yes, the curried form allows for efficient inline transform/chaining, for those that are over the learning curve.  There is nuance to be sure, but a couple of points push me to the camp of de-emphasizing currying in the code-base:

1) as mentioned, approachability is a major goal, which currying is a counter to

2) lodash_fp library no longer supported, and general "not natively supported by the language" makes advanced FP concepts like currying in js/ts harder to have good type flow and other safeguards and IDE awareness in place.

See where this has shaken out, I think doing the append inline without the append utility function is the winner in my book... it's lighter since it removed all the cruft/need for a library or utility completely.